src/elf.rs

```diff
@@ -1103,11 +958,6 @@ impl<C: ContextObject> Executable<C> {
                         addr = ebpf::MM_RODATA_START.saturating_add(addr);
                     }
 
-                    if text_section
-                        .file_range()
-                        .unwrap_or_default()
-                        .contains(&r_offset)
-                        || sbpf_version == SBPFVersion::V0
                     {
```

Comment:
I don't think the new scope here is necessary. Removing it improves readability. I'm approving the PR, since this shouldn't be a blocker.